lib/docs/main.js

```diff
@@ -642,14 +642,14 @@
         }
     }
 
-    function onHashChange(state) {
-      history.replaceState({}, "");
-      navigate(location.hash);
+    function handleNavigation(state, location_hash) {
+      history.replaceState("keepScrollPosition", "");
```

Comment:
I would say the string is *less* clear than the old code: this strongly implies that the string has semantic meaning, and it took me a second to understand that it doesn't. Instead, just document the intent with a comment:
```suggestion
      // Use a non-`null` state value to prevent the window scrolling if the user goes back to this history entry.
      history.replaceState({}, "");
```

Reply:
Sounds good to me.